app/models/user_export.rb

```diff
@@ -14,7 +15,14 @@ def ignore_missing_post_uploads
 
   def self.remove_old_exports
     UserExport.where('created_at < ?', 2.days.ago).find_each do |user_export|
-      user_export.destroy!
+      UserExport.transaction do
+        begin
+          Post.where(topic_id: user_export.topic_id).find_each { |p| p.destroy! }
```

Comment:
Hmm I think we should be careful here. If `user_export.topic_id` is `nil`, the query will find all posts that has `topic_id IS NULL`. That is though impossible since `Post#topic_id` is not `nullable`. Perhaps an `if` check to ensure that we don't run an extra query unnecessarily?

Reply:
Agreed, done.